aspnetcore/blazor/routing.md

```diff
@@ -37,7 +35,14 @@ In a Blazor server-side or client-side app:
 </Router>
 ```
 
-When a *.razor* file with an `@page` directive is compiled, the generated class is provided a <xref:Microsoft.AspNetCore.Mvc.RouteAttribute> specifying the route template. At runtime, the router looks for component classes with a `RouteAttribute` and renders the component with a route template that matches the requested URL.
+When a *.razor* file with an `@page` directive is compiled, the generated class is provided a <xref:Microsoft.AspNetCore.Mvc.RouteAttribute> specifying the route template.
+
+At runtime, the `RouteView` component:
+
+* Receives the `RouteData` from the `Router` along with any desired parameters.
+* Searches for component classes with a `RouteAttribute` and renders the component with a route template that matches the requested URL.
```

Comment:
The `RouteView` component doesn't search for components. The `RouteData` passed to the `RouteView` from the `Router` already specifies a specific component and the matched parameters. All the `RouteView` does is render the specified component with its layout (or an optional default layout) using the specified parameters.